File: BehavioralDesignPattern/MediatorPattern/python/ConcreteMediator/Chatgroup.py
from AbstractMediator.Mediator import Mediator


class Chatgroup(Mediator):
    def __init__(self,name):
        self.name = name
        self.users = []

    def add(self,user):
        if user not in self.users:
            self.users.append(user)

    def send(self,message,user):
        for u in self.users:
            if user != u:
                u.receive(message)

File: BehavioralDesignPattern/MediatorPattern/java/ConcreteMediator/ChatGroup.java
package BehavioralDesignPattern.MediatorPattern.java.ConcreteMediator;

import java.util.ArrayList;
import java.util.List;

import BehavioralDesignPattern.MediatorPattern.java.AbstractColleague.Colleague;
import BehavioralDesignPattern.MediatorPattern.java.AbstractMediator.Mediator;

public class ChatGroup implements Mediator{
    List<Colleague> users;
    String name;
    
    public ChatGroup(String name){
        this.users = new ArrayList<>();
        this.name = name;
    }

    public void add(Colleague colleague){
        if(!this.users.contains(colleague)){
            this.users.add(colleague);
        }
    }

    public void sendMessage(String message, Colleague colleague){
        for(Colleague user: users){
            if(user.getName()!=colleague.getName()){
                user.receive(message);
            }
        }
    }

}
